src/api/objects/ObjectAPI.js

```diff
@@ -226,7 +226,28 @@ define([
                     (identifier.namespace === identifiers[0].namespace &&
                         identifier.key === identifiers[0].key);
             });
-    }
+    };
+
+    ObjectAPI.prototype.getOriginalPath = function (identifier) {
```

Comment:
I wonder if we can simplify this even further? - 
``` JavaScript
getOriginalPath(identifier, path = []) {
    return this.get(identifier).then((domainObject) => {
        path.push(domainObject);
        if (domainObject.location) {
            return this.getOriginalPath(utils.parseKeyString(domainObject.location), path);
        } else {
            return path;
        }
    });
}
```

Reply:
We should use the same convention we have used with paths elsewhere (selection, navigation, etc.) going from child to parent, where the child is at array index 0, its parent at 1, etc. up the chain.

Reply:
I've also used `parseKeyString` in my example because I think that in old style object models, the location will be a keystring.